File: freight-triage-polyglot-ledger-recovery/environment/reconcile/internal/normx/lower_ascii.go
package normx

// LowerAscii normalizes with the lower ascii rule.
func LowerAscii(text string) string {
	out := []byte(text)
	for i, c := range out {
		if c >= 'A' && c <= 'Z' {
			out[i] = c + 32
		}
	}
	return string(out)
}

File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/norm/LowerAscii.java
package com.freight.norm;

/** Normalizer: lower ascii. */
public final class LowerAscii implements Normalizer {

    @Override
    public String name() {
        return "lower_ascii";
    }

    @Override
    public String apply(String text) {
        StringBuilder out = new StringBuilder(text.length());
        for (int i = 0; i < text.length(); i++) {
            char c = text.charAt(i);
            out.append((c >= 'A' && c <= 'Z') ? (char) (c + 32) : c);
        }
        return out.toString();
    }
}
